File: web/src/test/factories/project.ts
import type { Project } from "@/types";
import { FACTORY_GIT_REPO_ID } from "./git";

export const FACTORY_REPO_DEFAULT_PROJECT_ID =
  "00000000-0000-4000-8000-000000000040";

export function repoDefaultProjectFactory(
  overrides: Partial<Project> = {},
): Project {
  return {
    id: FACTORY_REPO_DEFAULT_PROJECT_ID,
    name: "Default",
    description: "",
    status: "active",
    context_summary: "",
    repository_id: FACTORY_GIT_REPO_ID,
    is_default: true,
    created_at: "2026-06-22T12:00:00Z",
    updated_at: "2026-06-22T12:00:00Z",
    ...overrides,
  };
}

export function projectFactory(overrides: Partial<Project> = {}): Project {
  return {
    id: "11111111-1111-4111-8111-111111111111",
    name: "Custom project",
    description: "",
    status: "active",
    context_summary: "",
    repository_id: FACTORY_GIT_REPO_ID,
    is_default: false,
    created_at: "2026-06-22T12:00:00Z",
    updated_at: "2026-06-22T12:00:00Z",
    ...overrides,
  };
}

File: pkgs/projects/store/model/project.go
package model

import (
	"time"

	"github.com/AlexsanderHamir/Hamix/pkgs/projects/domain"
)

// Project is the GORM persistence shape for domain.Project (columns only).
type Project struct {
	ID             string               `gorm:"primaryKey"`
	Name           string               `gorm:"not null;index"`
	Description    string               `gorm:"type:text;not null;default:''"`
	Status         domain.ProjectStatus `gorm:"not null;index;default:active;check:chk_projects_status,status IN ('active','archived')"`
	ContextSummary string               `gorm:"type:text;not null;default:''"`
	RepositoryID   *string              `gorm:"index"`
	IsDefault      bool                 `gorm:"not null;default:false;index"`
	// NextTaskNumber is the next per-project task.number to allocate (starts at 1).
	// Not exposed on the project HTTP API — persistence-only counter.
	NextTaskNumber int       `gorm:"not null;default:1"`
	CreatedAt      time.Time `gorm:"not null;index"`
	UpdatedAt      time.Time `gorm:"not null;index"`
}

// TableName pins the projects table name.
func (Project) TableName() string { return "projects" }
